### PR TITLE
feat: Hide link text if only one link and embedded - MEED-4546 - Meeds-io/MIPs#124

### PR DESCRIPTION
**Pull request by SaraBoutej**
This PR allows to hide the link text in activity body if there is a single link and embedded.

### CODE REVIEW

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/content/ActivityBody.vue [hunk 1]
```diff
@@ -103,6 +103,17 @@ export default {
...
+          this.body.indexOf('href') !== -1 &&
+          this.body.indexOf('href') === this.body.lastIndexOf('href')) {
+        const hasEmbeddedHtml = this.activity && this.activity.templateParams && this.activity.templateParams.html;
+        if (hasEmbeddedHtml) {
+          const startLinkIndex =  this.body.indexOf('<a href');
+          const endLinkIndex =  this.body.indexOf('</a>') + ('</a>').length;
+          const link = this.body.substring(startLinkIndex, endLinkIndex);
+          this.body = this.body.replaceAll(link, '');
+        }
+      }
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/content/ActivityBody.vue]: Why this change? This is very impactant in term of performances if all activity texts should be processed

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/ActivityStreamActivity.vue [hunk 2]
```diff
@@ -235,12 +235,12 @@ export default {
...
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityReplyToComment'
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'EditComment'
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'GamificationActionAnnouncedNotification';
+        const isLikeAction = this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'Like'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'LikeComment';
+        const isNewCommentAction = this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityComment'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityReplyToComment'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'EditComment'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'GamificationActionAnnouncedNotification';
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/ActivityStreamActivity.vue]: Please use extensionRegistry extensibility to determine which behavior to adopt for each activity type instead of embedding business logic of other addons in social which makes the code unmaintainable and not flexible enough for other DAO members (eXo addons)

Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/ActivityStreamActivity.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/content/ActivityBody.vue (hunk 1 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/content/ActivityBody.vue]: We need to check if the activity content text contains a single embedded link. If so, we should hide the link text from the activity content

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/ActivityStreamActivity.vue [hunk 3]
```diff
@@ -235,14 +238,13 @@ export default {
...
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'GamificationActionAnnouncedNotification';
-        this.isRead = this.unreadMetadata && !isLikeAction && !isNewCommentAction;
-        this.hasNewComment = this.unreadMetadata && isNewCommentAction;
+        const isLikeAction = this.unreadMetadata && this.unreadMetadata?.properties?.actionType === 'Like'
+            || this.unreadMetadata && this.unreadMetadata?.properties?.actionType === 'LikeComment';
+        const isNewActivityCommentAction = this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityComment'
+            || this.unreadMetadata && this.unreadMetadata?.properties?.actionType === 'ActivityReplyToComment'
+            || this.unreadMetadata && this.unreadMetadata?.properties?.actionType === 'EditComment';
+        this.isRead = this.unreadMetadata && !isLikeAction && !isNewActivityCommentAction;
+        this.hasNewComment = this.unreadMetadata && isNewActivityCommentAction || this.activityCommented;
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/ActivityStreamActivity.vue:247]: Do you mean?

```suggestion
        this.hasNewComment = this.unreadMetadata && (isNewActivityCommentAction || this.activityCommented);
```

**Review by boubaker — APPROVED** [review]

**Comment by Christyempire** [comment]
How are we getting the embedded link
